### ISSUE TITLE
CSM merge does not consider assigned data

### ISSUE THREAD

**Issue by vhirtham** [BAMWelDX/weldx#491]
As discussed with @marscher last Friday, I tested if merging two CSMs transfers data to the child CSM to the parent CSM:

~~~ python
from weldx import CoordinateSystemManager

csm_1 = CoordinateSystemManager("r")
csm_1.create_cs("a", "r", coordinates=[1, 3, 4])
csm_1.create_cs("b", "r", coordinates=[5, 5, 5])

csm_2 = CoordinateSystemManager("x")
csm_2.create_cs("a", "x", coordinates=[0, 3, 0])
csm_2.create_cs("y", "x", coordinates=[5, 0, 0])

csm_2.assign_data([[1, 2, 3], [4, 5, 6]], "data", "y")

print(csm_2.data_names)

csm_1.merge(csm_2)

print(csm_1.data_names)

unmerged = csm_1.unmerge()
print(unmerged[0].data_names)
~~~

output:

~~~
['data']
[]
[]
~~~

As you can see, this is not the case (unmerging fails too). As far as I remember, this should be a minor implementation issue. The data itself should be copied, but it isn't registered in the internal structure. I will check if unmerging is also affected.

**Comment by CagtayFabry** [comment]
does it make sense to look into that with #354 or do we need this right now?

**Comment by vhirtham** [comment]
> does it make sense to look into that with #354 or do we need this right now?

They are not really related to each other and @marscher said it would be nice to use it in the dashboards. I also think that this should be solved for `0.5.0` so that our main features are working as intended by then.

**Comment by CagtayFabry** [comment]
yeah for sure, just thought about it since #354 is part of `0.5.0` anyway
Maybe we can have a talk about timeframes for implementation
